tests/lib/rules/camelcase.js

```diff
@@ -790,6 +794,25 @@ ruleTester.run("camelcase", rule, {
                 }
             ]
         },
+        {
+            code: "const {mem_var = 1} = obh",
+            options: [{ ignoreDestructuring: false }],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{
+                messageId: "notCamelCase",
+                data: { name: "mem_var" },
+                type: "Identifier"
+            }]
+        },
+        {
+            code: "var obj = { mem_var: 1 };",
+            options: [{}],
+            errors: [{
+                messageId: "notCamelCase",
+                data: { name: "mem_var" },
+                type: "Identifier"
+            }]
+        },
```

Comment:
the test looks good to have, but not to test the change?

Reply:
this issue was originated because `babel/camelcase` has less option than the base rule so it was mutating with the option of the base rule while validating. So I can try integration testing with `eslint-plugin-babel` which will need this package as devdependency. 
Any other suggestions? cause with core rule, it seems working fine. It would be great to check with the core rules itself re-create this case. any input would be appreciated 💯 !

Reply:
I see,  can you add a test like [this comment](https://github.com/eslint/eslint/issues/12592#issuecomment-557825506)?

Reply:
after testing the above mentioned config (which was the original issue) with the current change implemented, the issue still exists. 

![image](https://user-images.githubusercontent.com/26347874/80307107-be1baf00-87e4-11ea-9168-38a89cdb0c57.png)


So, I think the current change (PR) is not correct change. 

After I cloned the rule option here in [config-validator](https://github.com/eslint/eslint/blob/master/lib/shared/config-validator.js#L186)

like this 

```js
 Object.keys(rulesConfig).forEach(id => {
        const rule = getAdditionalRule(id) || BuiltInRules.get(id) || null;
        const clonedRulesConfigForId = _.cloneDeep(rulesConfig[id])
        validateRuleOptions(rule, id, clonedRulesConfigForId, source);
    });
```

it is passing the test 

![image](https://user-images.githubusercontent.com/26347874/80307196-37b39d00-87e5-11ea-86dc-ee9a7e70cb19.png)

Reply:
cc @kaicataldo @aladdin-add should I implement the change mentioned above ?

Reply:
can you refactor the code to: cloning the config after loading?

Reply:
you meant somewhere [here](https://github.com/eslint/eslint/blob/fdfa364c1b0d88689d02cbe8ae848a3e323209aa/lib/cli-engine/config-array-factory.js#L492) , in this case we need to send the clonedconfig to the normalizer which will return the mutated clonedconfig and that mutated one will be returned from here. so, IMO this might not be the correct place. 

let me push the latest commit.

Reply:
@aladdin-add can you check the latest commit!

I think the current change is working as expected in order to fix the issue